### DISCUSSION TITLE
Secrets created using createOrUpdateRepoSecret  are not accessible

### DISCUSSION THREAD

**Discussion by lakshmajee** [octokit/octokit.js#2259] [Q&A]
- I have two workflows where one will create a GitHub secret using https://octokit.github.io/rest.js/ `createOrUpdateRepoSecret` mehod. This step adds secrets to my GitHub actions. When I printed it, it was there.
- I have another workflow where I need to pass this secret (created by the above workflow). Here the secret value is not readable.

More details 
https://github.com/actions/github-script/issues/276#issue-1309418335

I am not sure about the right place to ask help for with this issue. Could someone help?

Thanks in advance 🙏 .

**Comment by timrogers** [comment]
👋 Sorry to hear you're having some issues here! 

Does the repo secret you create show up in the GitHub UI, or is it missing there too?

Could you try reproducing the issue outside of Actions, for example running the JS on your own local machine? It'll be easier to inspect what is going on that way.

**Reply by lakshmajee** [reply]
Thanks, @timrogers 

Yes, secrets created using `createOrUpdateRepoSecret` are showing up in Github UI (actions secrets management)
Yes, will create a sample reproducible repository. (give me some time)  
Edit : Original issue https://github.com/actions/github-script/issues/276#issue-1309418335

**Reply by lakshmajee** [reply]
@timrogers Created a repository [here](https://github.com/awkward-minion/issue-gh-actions-rest-secets) which reproduces the issue

**Reply by timrogers** [reply]
I forked your repo and there seemed to be some issues with the workflow files, for example incorrect syntax and not passing the token to `@actions/github-script` in the right way. Can you try my new code at https://github.com/timrogers/issue-gh-actions-rest-secets? I can't test it properly as I don't have AWS credentials.

**Reply by lakshmajee** [reply]
@timrogers Merged your changes into mine and run the workflows, below are the result screenshots 
![secrets is visible in UI](https://user-images.githubusercontent.com/15211916/180431678-5bed3424-50cc-4c77-887c-af044d6d9216.png)
Error log for action https://github.com/awkward-minion/issue-gh-actions-rest-secets/runs/7467554686?check_suite_focus=true

**Reply by timrogers** [reply]
I get the same error. Is that error not because the value is bad though? What would happy if you gave real configuration that AWS accepts?

**Reply by lakshmajee** [reply]
~It is a real configuration,~ give me some time. Maybe I will make it a private repo for some time and add you as a collaborator is that fine ?

I have made some changes to the repo to replicate exact issue, now we see different errors
- With [Manual secret workflow](https://github.com/awkward-minion/issue-gh-actions-rest-secets/runs/7471463781?check_suite_focus=true), `aws-actions` were able to detect the `assume-role` inputs but error out with insufficient permissions (working as expected)
    ```bash
    Error: Not authorized to perform sts:AssumeRoleWithWebIdentity
    ```
- With [API secret workflow](https://github.com/awkward-minion/issue-gh-actions-rest-secets/runs/7471464986?check_suite_focus=true), `aws-actions` were unable to detect the `assume-role` input
    ```bash
    Error: Source Account ID is needed if the Role Name is provided and not the Role Arn.
    ```
